contents/docs/how-posthog-works/ingestion-pipeline.mdx

```diff
@@ -26,6 +26,11 @@ graph TD
 
     Kafka2 <-..- ClickHouse
 ```
+## Why am I seeing duplicate events?
+
+We recommend sending a UUID with every request, and some of our libraries such as `posthog-js` will do this automatically. If you don't send UUIDs for every event, we wouldn't be able to know it's the same event and hence we wouldn't be able to unduplicate it for you. Events with the same uuid, event name, timestamp and distinct_id are considered duplicates and will be eventually unduplicated. Keep in mind that failures and retries happen, so your application or our library might send the same event multiple times, which if you don't set UUIDs would end up in PostHog as separate events.
```

Comment:
```suggestion
We recommend sending a `uuid` value with every captured event. Events with the same UUID, event name, timestamp, and `distinct_id` are considered duplicates and are eventually de-duplicated. 

This is important because failures and retries happen, so your application or our library might send the same event multiple times. If you don't send UUIDs for every event, we aren't able to know if it's the same event, and hence we are not able to de-duplicate it for you.  
```

Reply:
I would still call out that posthog-js does it automatically and some other libs too 🤷‍♀️  ... we should just make all our libs do it, but who has time for that 😅

Reply:
> I would still call out that posthog-js does it automatically and some other libs too 🤷‍♀️ ... we should just make all our libs do it, but who has time for that 😅

I moved that down a paragraph (but should have just edited it all as one comment).